bookdown/08-interpretation.Rmd

```diff
@@ -1,10 +1,165 @@
+
+
+```{r learn, message=FALSE, warning=FALSE}
+learner = mlr_learners$get("classif.ranger")
+learner$predict_type = "prob"
+learner$train(task_peng)
+learner$model
+x = penguins[which(names(penguins) != "species")]
+model = Predictor$new(learner, data = x, y = penguins$species)
+```
+
+As explained in section [2.3](https://mlr3book.mlr-org.com/learners.html), specific learners can be queried with `mlr_learners`. In section [2.5](https://mlr3book.mlr-org.com/train-predict.html) it is recommended for some classifiers to use the `predict_type` as `prob` instead of directly predicting a label. This is what is done in this example. `penguins[which(names(penguins) != "species")]` is the data of all the features and `y` will be the penguins`species`. `learner$train(task_peng)` trains the model and `learner$model` stores the model from the training command. `Predictor` holds the the machine learning model and the data. All interpretation methods in `iml` need the the machine learning model and the data to be wrapped in the `Predictor` function. 
+
+Next is the core of the offerings from `iml`. In this example three separate interpretation methods will be used: [FeatureEffects](https://github.com/christophM/iml/blob/master/R/FeatureEffects.R), [FeatureImp](https://github.com/christophM/iml/blob/master/R/FeatureImp.R) and [Shapley](https://github.com/christophM/iml/blob/master/R/Shapley.R)
+
+
+* `FeatureEffects` computes the effects for all given features on the model prediction. Different methods are implemented: [Accumulated Local Effect (ALE) plots](https://christophm.github.io/interpretable-ml-book/ale.html), [Partial Dependence Plots (PDPs)](https://christophm.github.io/interpretable-ml-book/pdp.html) and [Individual Conditional Expectation (ICE) curves](https://christophm.github.io/interpretable-ml-book/ice.html).
+
+* `FeatureImp` computes the importance of features by calculating the increase in the model's prediction error after permuting the feature (more [here](https://christophm.github.io/interpretable-ml-book/feature-importance.html#feature-importance)).
+
+* `Shapley` computes feature contributions for single predictions with the Shapley value -- an approach from cooperative game theory ([Shapley Value](https://christophm.github.io/interpretable-ml-book/shapley.html)).
+
+
+**FeatureEffects**
+
+In addition to the commands above the following two need to be ran:
+
+```{r eff, message=FALSE, warning=FALSE, fig.cap='Plot of the results from FeatureEffects. FeatureEffects computes and plots feature effects of prediction models',  fig.align='center'}
+effect = FeatureEffects$new(model)
+effect$plot(features = c("bill_length_mm","bill_depth_mm", "flipper_length_mm", "body_mass_g", "body_mass_g","year"))
+```
+
+`effect` defines the interpretation model by wrapping `Predictor`. After this is done the selected features can be plotted. In this example, all of the features provided by the `palmerpenguins` data set were used. 
```

Comment:
The first sentence makes no sense. Please write something like "`effect` stores the resulting object of the feature effect computation."